internal/resource/config/authenticationpolicies/policy/authentication_policies_policy_resource.go

```diff
@@ -0,0 +1,252 @@
+	state.HandleFailuresLocally = types.BoolPointerValue(r.HandleFailuresLocally)
+
+	state.AuthenticationApiApplicationRef, respDiags = resourcelink.ToState(ctx, r.AuthenticationApiApplicationRef)
+	diags.Append(respDiags...)
+
+	state.RootNode, respDiags = authenticationpolicytreenode.ToState(ctx, r.RootNode, 1)
+	diags.Append(respDiags...)
+
+	return diags
+}
+
+func addOptionalAuthenticationPolicyFields(addRequest *client.AuthenticationPolicyTree, plan authenticationPoliciesPolicyModel) error {
+	addRequest.Id = plan.PolicyId.ValueStringPointer()
+	addRequest.Name = plan.Name.ValueStringPointer()
+	addRequest.Description = plan.Description.ValueStringPointer()
+	addRequest.Enabled = plan.Enabled.ValueBoolPointer()
+	addRequest.HandleFailuresLocally = plan.HandleFailuresLocally.ValueBoolPointer()
+
+	var err error
+	addRequest.AuthenticationApiApplicationRef, err = resourcelink.ClientStruct(plan.AuthenticationApiApplicationRef)
+	if err != nil {
+		return err
+	}
+
+	if internaltypes.IsDefined(plan.RootNode) {
+		addRequest.RootNode, err = authenticationpolicytreenode.ClientStruct(plan.RootNode)
+		if err != nil {
+			return err
+		}
+	}
+
+	return err
+}
+
+func (r *authenticationPoliciesPolicyResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan authenticationPoliciesPolicyModel
+
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	newPolicyTree := client.NewAuthenticationPolicyTree()
+	err := addOptionalAuthenticationPolicyFields(newPolicyTree, plan)
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to add optional properties to add request for the Authentication Policy", err.Error())
+		return
+	}
+
+	apiCreatePolicy := r.apiClient.AuthenticationPoliciesAPI.CreatePolicy(config.ProviderBasicAuthContext(ctx, r.providerConfig))
+	apiCreatePolicy = apiCreatePolicy.Body(*newPolicyTree)
+	policyResponse, httpResp, err := r.apiClient.AuthenticationPoliciesAPI.CreatePolicyExecute(apiCreatePolicy)
+	if err != nil {
+		config.ReportHttpError(ctx, &resp.Diagnostics, "An error occurred while creating the Authentication Policy", err, httpResp)
+		return
+	}
+
+	// Read the response into the state
+	var state authenticationPoliciesPolicyModel
+	diags = readAuthenticationPolicyResponse(ctx, policyResponse, &state)
+	resp.Diagnostics.Append(diags...)
+	diags = resp.State.Set(ctx, state)
+	resp.Diagnostics.Append(diags...)
+}
+
+func (r *authenticationPoliciesPolicyResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state authenticationPoliciesPolicyModel
+
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	policyResponse, httpResp, err := r.apiClient.AuthenticationPoliciesAPI.GetPolicy(config.ProviderBasicAuthContext(ctx, r.providerConfig), state.Id.ValueString()).Execute()
```

Comment:
This method should use PolicyId, the normal Id is only computed for terraform testing purposes.